fourcc.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+import sys
+import re
+
+fourcc_include = sys.argv[1]
+
+fmt_list = []
+mod_list = []
+
+fmt = re.compile(r'^#define (\w+)\s*(?:\\$\s*)?fourcc_code', flags=re.M)
+mod = re.compile(r'^#define (\w+)\s*(?:\\$\s*)?fourcc_mod_code', flags=re.M)
+
+def add_ident(l, ident_list):
+	parts = l.split()
+	ident = parts[1]
+	ident_list.append(ident)
+
+with open(fourcc_include) as f:
+	for l in f.readlines():
+		ident = fmt.search(l)
+		if ident != None:
+			add_ident(ident.group(0), fmt_list)
+		ident = mod.search(l)
+		if ident != None:
+			add_ident(ident.group(0), mod_list)
+
+f_name = sys.argv[2]
+
+with open(f_name, 'w') as f:
+	f.write('''\
+#include <stdint.h>
+#include <drm_fourcc.h>
+
+static inline const char *format_str(uint32_t format)
+{
+	switch (format) {
+		case DRM_FORMAT_INVALID:
+			return "INVALID";
+''')
+	for ident in fmt_list:
+		f.write('\t\tcase ' + ident + ':\n\t\t\treturn "' + ident[11:] + '";\n')
```

Comment:
I think this would look a lot cleaner as a format string.
```python
f.write('\t\tcase {}: return "{}";\n'.format(ident, ident[11:])
```
Also, I'm not a huge fan of the seemingly magic number 11. `len('DRM_FORMAT_')` is more obvious.